src/rsvc/srv_common.c

```diff
@@ -238,15 +238,24 @@ ds_rsvc_get_attr(struct ds_rsvc *svc, struct rdb_tx *tx, rdb_path_t *path,
 
 		rc = rdb_tx_lookup(tx, path, &key, &iovs[j]);
 
-		if (rc != 0) {
-			D_CDEBUG(rc == -DER_NONEXIST, DB_ANY, DLOG_ERR,
-				 "%s: failed to lookup attribute '"DF_KEY"': "
-				 DF_RC"\n",
-				 svc->s_name, DP_KEY(&key), DP_RC(rc));
+		if (rc == -DER_NONEXIST) {
+			/* attribute do not exist */
+			iovs[j].iov_buf_len = sizes[i];
+			iovs[j].iov_len = 0;
+			sizes[i] = 0;
+			/* fake crt_sgl_valid() */
+			iovs[j].iov_buf = (void *)(-1);
+
+			D_DEBUG(DB_ANY, "%s: failed to lookup attribute '"DF_KEY"': "DF_RC"\n",
```

Comment:
minor: instead of "failed" maybe just non-existent attr.

Reply:
Well this was the original msg even for non-existent case ... Tell me if you really want this to be changed anyway.

Reply:
no that was just minor. only if you are going to update / repush for some reason.

Reply:
I agree the wording here could be improved, in fact I changed the previous message to use D_CDEBUG so it wouldn't log errors on normal cases.

Another issue here is '"DF_KEY"', the DF_KEY does not simply print the string so using it with extra 's around is not required, in fact it probably makes the logs worse.

Reply:
> I agree the wording here could be improved, in fact I changed the previous message to use D_CDEBUG so it wouldn't log errors on normal cases.

well why did you not use two separate/different msgs/wording already ?

> Another issue here is '"DF_KEY"', the DF_KEY does not simply print the string so using it with extra 's around is not required, in fact it probably makes the logs worse.

But you seemed to have used the same already, as I just re-used existing syntax!

Reply:
Did I?  It was a mistake if I did.  Happy to land this though and we can address later.

Reply:
```
bfaccini$ git blame src/rsvc/srv_common.c
......................
b29d1fbb186 (Ashley Pittman  2021-05-19 17:49:15 +0100 242)                     D_CDEBUG(rc == -DER_NONEXIST, DB_ANY, DLOG_ERR,
b29d1fbb186 (Ashley Pittman  2021-05-19 17:49:15 +0100 243)                              "%s: failed to lookup attribute '"DF_KEY"': "
b29d1fbb186 (Ashley Pittman  2021-05-19 17:49:15 +0100 244)                              DF_RC"\n",
b29d1fbb186 (Ashley Pittman  2021-05-19 17:49:15 +0100 245)                              svc->s_name, DP_KEY(&key), DP_RC(rc));
......................
```